src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -351,6 +360,7 @@ public SegmentMetadata.ComponentMetadataMap flush(IndexDescriptor descriptor, Se
         try (var pqOutput = IndexFileUtils.instance.openOutput(descriptor.fileFor(IndexComponent.PQ, context), true);
              var postingsOutput = IndexFileUtils.instance.openOutput(descriptor.fileFor(IndexComponent.POSTING_LISTS, context), true);
              var indexWriter = new OnDiskGraphIndexWriter.Builder(builder.getGraph(), indexFile.toPath())
+                               .withVersion(JVECTOR_2_VERSION) // always write old-version format since we're not using the new features
                                .withMap(finalOrdinalMap)
                                .with(new InlineVectors(vectorValues.dimension()))
                                .withStartOffset(termsOffset)
```

Comment:
Looks like there is a similar `seek` a few lines below this. I think we can remove it like we did in the CompactionGraph.

Reply:
good catch!

Reply:
... turns out both of these seeks are in fact necessary -- the jvector writer will seek internally as needed but we still need to do this one explicitly so that we write the SAI segment header in the correct place

Reply:
added a comment to clarify this